### ISSUE TITLE
testing-library/prefer-find-by doesn't seem to be working

### ISSUE THREAD

**Issue by cuongluu8** [testing-library/eslint-plugin-testing-library#406]
I have it enabled in my `eslintrc.js` config file.

```js
{
  overrides: [
    {
      "files": ["*.test.js"],
      "extends": ["plugin:testing-library/react"],
      rules: {
        "testing-library/await-async-query": "error",
        "testing-library/no-await-sync-query": "error",
        "testing-library/no-debug": "error",
        "testing-library/prefer-find-by": "error",
        "testing-library/no-node-access": "off",
      }
    },
  ]
}
```

```js
await waitFor(() => {
  screen.getByLabelText("Name");
});
```

The above code doesn't seen to be coming up with an error.

Using v4.9.0

**Comment by Belco90** [comment]
Hello @cuongluu8!

Since you are restricting `eslint-plugin-testing-library` to `"*.test.js"` files, could you share the path and name of the file where the error is not reported? Are other `eslint-plugin-testing-library` problems reported in that file?

**Comment by cuongluu8** [comment]
> Hello @cuongluu8!
> 
> Since you are restricting `eslint-plugin-testing-library` to `"*.test.js"` files, could you share the path and name of the file where the error is not reported? Are other `eslint-plugin-testing-library` problems reported in that file?

Yes, changing `testing-library/no-node-access` to `error` will report errors in the same test file, so I know my setup is working

**Comment by Belco90** [comment]
Ok, I think I just found out why

```
await waitFor(() => {
  // not returning anything
  screen.getByLabelText("Name");
});
```

The callback you are passing to `waitFor` is not returning anything. I'd say from memory the rule only reports queries returned from the callback. Can you try returning the query? 

That's potentially an issue since you need to return something to be awaited.

**Comment by cuongluu8** [comment]
you're right!

But I'm thinking this rule should pick this up?  Also why playing around I'm also thinking this rules should error on this
```
// error
await waitFor(() =>
    expect(screen.getByText("test name")).toBeInTheDocument()
);

// ok
expect(await screen.findByText("test name")).toBeInTheDocument();
```
This is based on https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-waitfor-to-wait-for-elements-that-can-be-queried-with-find

> Advice: use find* any time you want to query for something that may not be available right away.

**Comment by Belco90** [comment]
> But I'm thinking this rule should pick this up?

I guess that should be another rule, another ESLint plugin, or even caught with TS if possible.

> I'm also thinking this rules should error on this
> 
> // error
> await waitFor(() =>
>     expect(screen.getByText("test name")).toBeInTheDocument()
> );
> 
> // ok
> expect(await screen.findByText("test name")).toBeInTheDocument();

I'd say that's also out of the scope of this rule since there is an assertion inside `waitFor` and you might want that rather than using a `findBy*` query.

Closing the issue then since the original problem is solved.
